addons/account/models/partner.py

```diff
@@ -167,11 +167,17 @@ def get_fiscal_position(self, partner_id, delivery_id=None):
         # This can be easily overridden to apply more complex fiscal rules
         PartnerObj = self.env['res.partner']
         partner = PartnerObj.browse(partner_id)
+        delivery = PartnerObj.browse(delivery_id)
 
-        # if no delivery use invoicing
-        if delivery_id:
-            delivery = PartnerObj.browse(delivery_id)
-        else:
+        company = self.env.company
+        eu_country_codes = self.env.ref('base.europe').country_ids.mapped('code')
```

Comment:
Since you are checking twice the if an element is inside this list, it might make sense to cast it as a set

```suggestion
        eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
```

Reply:
Will do, but could you please elaborate ?
We're just checking that a code is inside the list, we're not adding it.
So I'm not sure I see the advantage of using a set instead :sweat_smile:

Reply:
Looking at the time complexity:
* `'a' in a_list` is `O(n)`
* `'a' in a_set` is `O(1)`
* `set(a_list)` is `O(n)`

What you are doing now is twice `'a' in a_list`, so `O(2*n)`.
What I propose to do is `set(a_list)` + twice `'a' in a_set`, so `O(n+2)`

For the space complexity:
* `'a' in a_list` is `O(1)`
* `'a' in a_set` is `O(1)`
* `set(a_list)` is `O(2n)`
* `a_list` is `O(n)`
What you are doing now is twice `'a' in a_list`, so `O(n + 2)`.
What I propose to do is `set(a_list)` + twice `'a' in a_set`, so `O(2*n + 2)`


__________

This is a bit simplified, but hey...
So what I propose is better for the time but worse for the space.
It should be twice as fast at the cost of consuming a bit more memory.

Note that `O(n)` is the same as `O(2n)` (linearly equivalent), so this is really just a minor detail, but it is worth having the reflex when coding :)

Reply:
OK I see! Thanks for the explanation :ok_hand: 
I didn't know there were such differences in terms of complexity between a set and a list :+1: